### ISSUE TITLE
Slightly Off Beaufort Scale Numbers

### ISSUE THREAD

**Issue by srwareham** [aukedejong/lovelace-windrose-card#9]
It seems in some instances the hardcoded values for Beaufort scale conversion are slightly off. When using just m/s, these are close enough to not really matter. When converting to mph or other units, though, these are off by a multiplier and move the boundaries in an unintuitive way.
https://github.com/aukedejong/lovelace-windrose-card/blob/9222416552efcd747b491cf50087168124925642/src/WindSpeedConverter.ts#L137-L152

Beaufort number | Wind speed | code value
-- | -- | --
0 | < 0.5 m/s | 0.3
1 | 0.5–1.5 m/s | 1.6
2 | 1.6–3.3 m/s | 3.4
3 | 3.4–5.5 m/s | 5.5
4 | 5.5–7.9 m/s | 8
5 | 8–10.7 m/s | 10.8
6 | 10.8–13.8 m/s | 13.9
7 | 13.9–17.1 m/s | 17.2
8 | 17.2–20.7 m/s | 20.8
9 | 20.8–24.4 m/s | 24.5
10 | 24.5–28.4 m/s | 28.5
11 | 28.5–32.6 m/s | 32.7
12 | ≥ 32.7 m/s | N/A

Sources for Beaufort Scale Values
* https://en.wikipedia.org/wiki/Beaufort_scale
* https://www.weather.gov/mfl/beaufort

**Comment by aukedejong** [comment]
The Dutch wikipedia page states 0 - 0.2 m/s instead of 0.5 m/s.  The Dutch meteo institute also uses 0.2 m/s. Strange, other sites use 0.5.  Is it country specific?

About the other ranges. In code the second number is the same as the start of the next range.
The matching works like this:
- speed >= 0.5  and < 1.5
- speed >= 1.5 and < 3.4
This way there are no gaps.

I will change the 0.3 to 0.5, but I think the others are correct.

When the input unit is not meters per second, it is converted to m/s. Unless I made an error in the unit converting code (wrong factor maybe).
When the input speed unit is not defined, meters per second is used. If that's not the correct input unit, the ranges wil not be correct.

I can't think of an other way how the boundaries can be off by a multiplier.
Can you give more details?

**Comment by srwareham** [comment]
TL;DR I don't think you should change things to a convention other than whatever suits you best--thanks again for the great product!

Looking into it more, it seems like 0.2 m/s might actually be the canonical value. Although, confusingly enough, sources don't seem to be internally consistent. The Beaufort scale wasn't defined in reference to actual wind speeds, but the seemingly most accepted conversion from Beaufort value to m/s is
`meters per second = 0.836 * B^(3/2) `

Tabulating that, the bounds would be 
![image](https://user-images.githubusercontent.com/3392952/226498344-47d5cdfb-4a3e-430c-bec8-2a4c07bdab0c.png)


Yet it seems each source somewhat arbitrarily decides which round numbers it likes more. So I guess maybe it's a moot point. My comment regarding the multiplier is also probably similarly moot. Basically, when Beaufort 4 is defined as 5.5 - 8 m/s, that translates to 12.303 - 17.895 mph. Whereas the (arbitrarily?) chosen round numbers typically used for mph are actually 13 - 18. So if someone is using the default Beaufort implementation, but also has some other reporting of their wind speeds, they'd be confused when the two don't reconcile. Or, in my case, when I tried to re-implement the Beaufort scale in MPH (so I could have mph labels), the two implementations produced surprisingly different results.

**Comment by aukedejong** [comment]
I think other people will run into this to.
And I also think it's an easy fix to use a Bft scale that depends on the input unit.
And then a conversion to mps is not needed anymore. Is a performance improvements for the larger data sets.

Thanks for this analysis.
